src/OrchardCore.Modules/OrchardCore.Queries/Sql/GraphQL/SqlQueryFieldTypeProvider.cs

```diff
@@ -10,42 +10,47 @@
 using Microsoft.AspNetCore.Http;
 using Microsoft.Extensions.DependencyInjection;
 using Microsoft.Extensions.Logging;
-using OrchardCore.Apis.GraphQL;
 using OrchardCore.Apis.GraphQL.Resolvers;
 using OrchardCore.ContentManagement.GraphQL.Queries;
+using OrchardCore.Entities;
+using OrchardCore.Queries.Indexes;
+using OrchardCore.Queries.Sql.Models;
+using YesSql;
 
 namespace OrchardCore.Queries.Sql.GraphQL.Queries
 {
     /// <summary>
-    /// This implementation of <see cref="ISchemaBuilder"/> registers
+    /// This implementation of <see cref="Apis.GraphQL.ISchemaBuilder"/> registers
     /// all SQL Queries as GraphQL queries.
     /// </summary>
-    public class SqlQueryFieldTypeProvider : ISchemaBuilder
+    public class SqlQueryFieldTypeProvider : Apis.GraphQL.ISchemaBuilder
     {
         private readonly IHttpContextAccessor _httpContextAccessor;
         private readonly ILogger _logger;
 
-        public SqlQueryFieldTypeProvider(IHttpContextAccessor httpContextAccessor, ILogger<SqlQueryFieldTypeProvider> logger)
+        public SqlQueryFieldTypeProvider(
+            IHttpContextAccessor httpContextAccessor,
+            ILogger<SqlQueryFieldTypeProvider> logger)
         {
             _httpContextAccessor = httpContextAccessor;
             _logger = logger;
         }
+
         public Task<string> GetIdentifierAsync()
-        {
-            var queryManager = _httpContextAccessor.HttpContext.RequestServices.GetService<IQueryManager>();
-            return queryManager.GetIdentifierAsync();
-        }
+            => Task.FromResult(string.Empty);
```

Comment:
This can't be just an empty string, because the GraphQL schema must be rebuild whenever a query changes. The same is true for all the other `ISchemaBuilders`. Also, this should be somehow be cached, because it runs whenever a GraphQL query is executed.

Reply:
@gvkries how do you suggest we generate the identifier here? I see other implementation are using empty string.

Reply:
The difference is that these GraphQL fields are dynamic. For each query an additional type gets generated and the schema cache is invalidated when the identifier changes. So the `ISchemaBuilders` for field types that do not change can return an empty identifier, but the queries cannot. I guess this is one of the reasons why the document manager was used in the first place.
I fear this change gets even more nasty...

Reply:
@sebastienros I am not too familiar with GraphQL implementation. Do you have a good idea on what should the key identifier be here? Do we need to query the latest updated query and use that as a key? Like Query_{DateTime} so the cache is reset when any query document change?

Reply:
Can we just use the entity ID, it looks like you always create a new row when a query is saved. But we should try to avoid having to query the database one more time for each GraphQL request.

Reply:
Query name is also unique. So maybe somehow we need to get the query name from the request and return it. I will have to dig in to see how to get the query name from the request.

Reply:
Just keep in mind that the identifier has to change if any query of a specific type has changed.

Reply:
> Just keep in mind that the identifier has to change if any query of a specific type has changed.

Any query type or when the query in the request changes? If the request is querying `RecentPosts` query, and the `RecentArticles` query was changes, do we have to create the cache for `RecentPosts` query? I would think only if `RecentPosts` query change, then we need to provide a new key, right?

Reply:
@gvkries check out the updated changes. Let me know if you see any issues with the approach I took

Reply:
I think your approach is good, it solves this problem.